server/src/main/java/org/elasticsearch/immutablestate/ImmutableClusterStateHandler.java

```diff
@@ -95,4 +93,18 @@ default void validate(MasterNodeRequest<?> request) {
             throw new IllegalStateException("Validation error", exception);
         }
     }
+
```

Comment:
Splitting off the parsing part of handling immutable cluster state, so that we can reuse the immutable cluster state update controller to work with plain Java objects. This will come in handy when plugins/modules will want to save state on initialization.